### PR TITLE
[MOS-686] Fixed MTP availability only after phone unlocked

### PR DESCRIPTION
**Pull request by rrandomsky**
Fixed file access via MTP even when phone is not unlocked. Now access is granted when the phone is unlocked by the user entering a passcode. If the phone is not passcode protected (passcode is nor set) then access to the files is always possible via MTP.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

File: module-apps/apps-common/locks/handlers/PhoneLockHandler.cpp [hunk 1]
```diff
@@ -127,12 +130,12 @@ namespace locks
         owner->bus.sendMulticast(std::make_shared<locks::UnlockedPhone>(), sys::BusChannel::PhoneLockChanges);
     }
 
-    void PhoneLockHandler::phoneInputRequiredAction()
+    void PhoneLockHandler::phoneInputRequiredAction(bool isSoftRequest = false)
```
Comment (lukasmas) [inline module-apps/apps-common/locks/handlers/PhoneLockHandler.cpp:133]: Don't use bool as a function parameter. 
I recommend splitting the function or using an enum instead. 
I will look better and will be easier to understand

Comment (rrandomsky) [reply module-apps/apps-common/locks/handlers/PhoneLockHandler.cpp]: Ok, you right.
I changed bool to enum `PhoneLockHandler::TypeOfRequest`

Maybe code is still not clear enough and I should split functions?

Comment (lukasmas) [reply module-apps/apps-common/locks/handlers/PhoneLockHandler.cpp]: I would consider renaming it "PhoneLockHandler::TypeOfRequest" 
As I noticed it has 2 values Soft and Normal and for example, TypeOfRequest::Soft is not saying much about what it is and what it does
Try something that will explain this -> why unlock for MTP is soft? (UnlockType:: {UnlockPhone, UnlockMTPOnly} 
`
connect(typeid(locks::UnlockPhone), [&](sys::Message *request) -> sys::MessagePointer {
            return phoneLockHandler.handleUnlockRequest(locks::PhoneLockHandler::TypeOfRequest::Normal);
        });
        connect(typeid(locks::UnlockPhoneForMTP), [&](sys::Message *request) -> sys::MessagePointer {
            return phoneLockHandler.handleUnlockRequest(locks::PhoneLockHandler::TypeOfRequest::Soft);
        });
`
if it doesn't help much try to consider the introduction of a second function dedicated only to unlocking MTP

File: module-apps/apps-common/locks/handlers/PhoneLockHandler.hpp [hunk 2]
```diff
@@ -20,6 +20,13 @@ namespace locks
 
     class PhoneLockHandler
     {
+      public:
+        enum class TypeOfRequest
+        {
+            Normal = 0,
+            Soft
```
Comment (SP2FET) [inline module-apps/apps-common/locks/handlers/PhoneLockHandler.hpp:27]: why do we have two types of phone lock?

File: module-apps/apps-common/locks/handlers/PhoneLockHandler.cpp (hunk 1 above)
Comment (rrandomsky) [reply module-apps/apps-common/locks/handlers/PhoneLockHandler.cpp]: Ok then
I changed _enum_ names to:
```cpp
        enum class ReasonForRequest
        {
            Unspecified = 0, // For default behaviors
            UnlockForMTP
        };
 ```
 and added a proper comment where passing of the _new enum_  as arg. actually changes the behavior:
 ```cpp
        // If it's for MTP, don't show the input passcode popup if it's already on the screen
        data->ignoreIfTheSamePopupIsOnTopOfTheStack = (reqReason == ReasonForRequest::UnlockForMTP);
```

File: module-apps/apps-common/locks/handlers/PhoneLockHandler.hpp (hunk 2 above)
Comment (rrandomsky) [reply module-apps/apps-common/locks/handlers/PhoneLockHandler.hpp]: Applies to the scenario where the user plugs in the USB cable when there is already a pop-up on the screen to enter the password.
This scenario was mentioned by QA ["After connecting Pure to the computer, while entering the passcode,[...]](https://appnroll.atlassian.net/browse/MOS-686?focusedCommentId=67376)

File: module-apps/apps-common/locks/handlers/PhoneLockHandler.cpp (hunk 1 above)
Comment (lukasmas) [reply module-apps/apps-common/locks/handlers/PhoneLockHandler.cpp]: > Unspecified = 0, // For default behaviors

So maybe default? :D

File: module-apps/apps-common/locks/handlers/PhoneLockHandler.hpp [hunk 3]
```diff
@@ -20,6 +20,13 @@ namespace locks
 
     class PhoneLockHandler
     {
+      public:
+        enum class ReasonForRequest
+        {
+            Default = 0, // For default behaviors
+            UnlockForMTP
```
Comment (SP2FET) [inline module-apps/apps-common/locks/handlers/PhoneLockHandler.hpp:27]: I'm wondering if the name is proper - AFAIU this case is when you connect the phone to the PC and MTP wants to unlock the phone - so maybe it should be named `MTPUnlock` or something?

**Review by SP2FET — APPROVED** [review]

**Review by marziel — APPROVED** [review]

**Pull request merged**